erts/etc/win32/erl.c

```diff
@@ -235,33 +235,51 @@ static wchar_t *find_erlexec_dir(wchar_t *erlpath)
     wchar_t *dir =_wcsdup(erlpath);
     wchar_t *p;
     wchar_t *p2;
+
+    wchar_t *long_dir;
+    DWORD length;
     
     /* Chop of base name*/
     for (p = dir+wcslen(dir)-1 ;p >= dir && *p != L'\\'; --p)
         ;
     *p =L'\0';
     p--;
 
+    length = GetLongPathNameW(dir, NULL, 0);
```

Comment:
We can get rid of `_wcsdup(erlpath);` if we chop off the base name after getting the long path name, saving us from having to `free(dir)` everywhere.